engine/access/ingestion2/latest_persisted_sealed_result_test.go

```diff
@@ -0,0 +1,308 @@
+package ingestion2
+
+import (
+	"fmt"
+	"sync"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/storage"
+	storagemock "github.com/onflow/flow-go/storage/mock"
+	"github.com/onflow/flow-go/utils/unittest"
+)
+
+// TestNewLatestPersistedSealedResult tests the initialization of LatestPersistedSealedResult.
+// It verifies that:
+// - The initializer is called with the correct height
+// - The returned ConsumerProgress is properly stored
+// - All fields are correctly initialized on success
+// - Errors from the initializer are properly propagated
+func TestNewLatestPersistedSealedResult(t *testing.T) {
+	height := uint64(100)
+
+	t.Run("successful initialization", func(t *testing.T) {
+		mockInitializer := storagemock.NewConsumerProgressInitializer(t)
```

Comment:
we could use real modules instead of mocks, so that we don't have to mock so many methods.

Reply:
that's fair. I used ai to generate these, but I can update to make them simpler